scripts/create_status_report.py

```diff
@@ -0,0 +1,231 @@
+import sys
+from datetime import datetime, timedelta
+import os
+import os.path
+
+STATUS_TARGET_NAME = 0
+STATUS_TARGET_LAST_COMLETED_DATE = 1
+STATUS_TARGET_STATUS = 2
+STATUS_TARGET_STATUS_DATE = 3
+STATUS_TARGET_DURATION = 4
+STATUS_TARGET_DEPENDANCIES = 5
+STATUS_TARGET_LOG_FOLDER = 6
+
+DATE_FORMAT = "%Y-%m-%dT%H:%M:%S"
+
+BUFFER_OF_DAYS = 15
+
+dashboard_file_name = "index.html"
+tblStatus = []
+
+# ====================================================================================
+# home-brew relational DB interface
+# plain files pipe (|) separated
+# ====================================================================================
+def loadDBFile(strInputFile, encoding="utf-8", separator="|"):
+    cells = []
+    if os.path.exists(strInputFile):
+        filehandle = open(strInputFile, 'r', encoding=encoding)
+        txt = filehandle.readline().strip()
+        while len(txt) != 0:
+            if txt[0:1] != "#":
+                row = txt.strip().split(separator)
+                for i in range(len(row)):
+                    row[i] = row[i].strip()
+                if len(row) > 1:
+                    cells.append(row)
+            txt = filehandle.readline()
+        # end of while
+        filehandle.close()
+    return cells
+
+def print_html(strOutputFile, tblStatus1):
+    tblStatus = sorted(
+        tblStatus1, key=lambda record: record[STATUS_TARGET_STATUS_DATE], reverse=True)
+
+    pipeline_status = 'Idle'
+    n_targets_total = 0
+    n_targets_in_progress = 0
+    n_targets_frozen = 0
+    n_targets_failed = 0
+    oldest_completed = ''
+
+    for record in tblStatus:
+        n_targets_total = n_targets_total + 1
+        if record[STATUS_TARGET_STATUS] == "started":
+            pipeline_status = 'Up and Running'
+            n_targets_in_progress = n_targets_in_progress + 1
+        elif record[STATUS_TARGET_STATUS] == "failed":
+            n_targets_failed = n_targets_failed + 1
+        elif record[STATUS_TARGET_STATUS] == "completed":
+            if oldest_completed == '':
+                oldest_completed = record[STATUS_TARGET_STATUS_DATE]
+            elif oldest_completed > record[STATUS_TARGET_STATUS_DATE]:
+                oldest_completed = record[STATUS_TARGET_STATUS_DATE]
+
+    fo = open(strOutputFile, 'w', encoding="utf-8")
+    fo.write("""
+        <html>
+
+        <head>
+            <link rel="stylesheet" href='report/style.css'>
+            <link href='https://fonts.googleapis.com/css?family=IBM Plex Sans Condensed' rel='stylesheet'>
+            <script src='report/sorttable.js' type="text/javascript"> </script>
+            <script src='report/script_u.js' type="text/javascript"> </script>
+        </head>
+
+        <body>
+            <div id="stuff">
+                <img class="img" src='report/Kontur_logo_main.png' />
```

Comment:
why kontur logo in mapaction project?

Reply:
geocint-runner _is_ mapaction project then

Reply:
They will be able to place their own logo. So it's not a problem.